core/src/main/java/org/polypheny/db/util/background/BackgroundTaskHandle.java

```diff
@@ -44,7 +45,12 @@ public BackgroundTaskHandle( String id, BackgroundTask task, String description,
 
         // Schedule
         ScheduledExecutorService exec = Executors.newSingleThreadScheduledExecutor();
-        this.runner = exec.scheduleAtFixedRate( this, 0, schedulingType.getMillis(), TimeUnit.MILLISECONDS );
+        if ( schedulingType.getDelayType() == TaskDelayType.FIXED ) {
+            this.runner = exec.scheduleAtFixedRate( this, 0, schedulingType.getMillis(), TimeUnit.MILLISECONDS );
+        } else {
```

Comment:
else if ( schedulingType.getDelayType() == TaskDelayType.DELAYED ) {

} else {
throw new ...
}

Reply:
else if ( schedulingType.getDelayType() == TaskDelayType.DELAYED ) {

 } else {
throw new RuntimeException("A not known TaskDelayType was used. Only TaskDelayType.FIXED and TaskDelayType.DELAYED are implemented.");
        }